core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt

```diff
@@ -36,14 +38,22 @@ abstract class BaseActivity : AppCompatActivity() {
 
   private var unbinder: Unbinder? = null
 
+  abstract fun injection(coreComponent: CoreComponent)
+
   override fun onCreate(savedInstanceState: Bundle?) {
-    injection()
+    CoreApp.getCoreComponent().inject(this)
```

Comment:
keep the invocation of `injection` and don't inject here, only concrete classes should perform injection